turbinia/workers/analysis/jenkins.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task for analysing Jenkins."""
+
+from __future__ import unicode_literals
+
+import os
+import re
+
+from turbinia import TurbiniaException
+from turbinia.evidence import ReportText
+from turbinia.workers import TurbiniaTask
+from turbinia.lib.utils import extract_artifacts
+from turbinia.lib.utils import bruteforce_password_hashes
+
+
+class JenkinsAnalysisTask(TurbiniaTask):
+  """Task to analyze a Jenkins install."""
+
+  def run(self, evidence, result):
+    """Run the Jenkins worker.
+
+    Args:
+        evidence (Evidence object):  The evidence to process
+        result (TurbiniaTaskResult): The object to place task results into.
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    # What type of evidence we should output.
+    output_evidence = ReportText()
+
+    # Where to store the resulting output file.
+    output_file_name = 'jenkins_analysis.txt'
+    output_file_path = os.path.join(self.output_dir, output_file_name)
+
+    # Set the output file as the data source for the output evidence.
+    output_evidence.local_path = output_file_path
+
+    try:
+      collected_artifacts = extract_artifacts(
+        artifact_names=['JenkinsConfigFile'],
+        disk_path=evidence.local_path,
+        output_dir=os.path.join(self.output_dir, 'artifacts')
+      )
+    except TurbiniaException as e:
+      result.close(self, success=False, status=str(e))
+      return result
+
+    version = None
+    credentials = []
+    for filepath in collected_artifacts:
+        with open(filepath, 'r') as input_file:
+          config = input_file.read()
+
+        extracted_version = self._extract_jenkins_version(config)
+        extracted_credentials = self._extract_jenkins_credentials(config)
+
+        if extracted_version:
+          version = extracted_version
+
+        credentials.extend(extracted_credentials)
+
+    analysis_report = self.analyze_jenkins(version, credentials)
+    output_evidence.text_data = analysis_report
+
+    # Write the report to the output file.
+    with open(output_file_path, 'w') as fh:
+      fh.write(output_evidence.text_data.encode('utf8'))
+      fh.write('\n'.encode('utf8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True)
```

Comment:
Can we add a 'status' here with a summary of what was discovered?  Maybe we could use the first line of the analysis report (possibly with some small changes)?